examples/lock-app/silabs/src/LockManager.cpp

```diff
@@ -558,13 +599,34 @@ DlStatus LockManager::GetWeekdaySchedule(chip::EndpointId endpointId, uint8_t we
     VerifyOrReturnValue(IsValidWeekdayScheduleIndex(weekdayIndex), DlStatus::kFailure);
     VerifyOrReturnValue(IsValidUserIndex(userIndex), DlStatus::kFailure);
 
-    const auto & scheduleInStorage = mWeekdaySchedule[userIndex][weekdayIndex];
-    if (DlScheduleStatus::kAvailable == scheduleInStorage.status)
+    // Get schedule data from nvm3
+    chip::StorageKeyName scheduleDataKey = LockUserWeekDayScheduleEndpoint(userIndex, weekdayIndex, endpointId);
+
+    uint16_t size = WeekDayScheduleInfoSize; // Create a non-const variable
+
+    error = mStorage->SyncGetKeyValue(scheduleDataKey.KeyName(), &weekDayScheduleInStorage, size);
```

Comment:
In all these places, should be checking that we did not get short reads....  I'll stop commenting on this.

Reply:
Added checks after each SyncGetKeyValue to check the size outparam